### ISSUE TITLE
Allow modes to override PdeParseTreeListener in PdePreprocessor

### ISSUE THREAD

**Issue by sampottinger** [benfry/processing4#11]
In talking with @codeanticode, it's not clear how modes should modify preprocessor behavior, Specifically, with the patterns introduced by ANTLR 4, the parse tree listener is where the preprocessing edits are generated and some client code (like modes) may need to override some of the preprocessing edit behavior.

**Comment by codeanticode** [comment]
@sampottinger thanks a lot for this! 

So the way to use the mode-specific preprocessor would be:

```
AndroidPreprocessor preproc = AndroidPreprocessorFactory.build(sketch.getName(), getPackageName());
```

correct?

I should also be able to similarly override the footer writing in the mode, shouldn't I?

**Comment by sampottinger** [comment]
Sorry! I think I responded by email but not here. For progeny, yes that's correct.

**Comment by sampottinger** [comment]
This is resolved with #13. Closing...
